api/src/main/java/at/helpch/chatchat/api/event/PMSendEvent.java

```diff
@@ -0,0 +1,104 @@
+package at.helpch.chatchat.api.event;
+
+import at.helpch.chatchat.api.Format;
+import java.util.Optional;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+public class PMSendEvent extends Event implements Cancellable {
+
+    private static final HandlerList HANDLERS = new HandlerList();
+    private boolean cancelled = false;
+
+    private @NotNull Optional<Component> cancelReason = Optional.of(
```

Comment:
im really not keen on this. I think it should be up to the API user to handle this & send the player a message as to why its cancelled.
If this is to stay, then this doesn't need to be wrapped in Optional anyway.

Reply:
Yeah I guess it would make more sense to let other plugins decide what to do (send message or not). I was thinking optional if other developers want to remove the message completely.